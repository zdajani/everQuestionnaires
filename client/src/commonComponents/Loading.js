import React from 'react';
import './styles/Loading.css';

const Loading = () => (
  <div className="loading-screen">
  	<img src={'https://tools.cisco.com/security/center/Resources/images/loader.gif'} alt="loadig" id="loading"/>
  </div>
);

export default Loading;