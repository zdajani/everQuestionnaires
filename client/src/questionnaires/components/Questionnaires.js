import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'
import QuestionnairesList from '../components/QuestionnairesList'
import './styles/Questionnaires.css'; 

const Questionnaires = ({ questionnaires, url }) => {
  const isAdminPage = url.includes('admin');
  return (
    <div>
      <section className="jumbotron text-center">
        <div className="container">
          <h1 className="jumbotron-heading page-title">{isAdminPage ? "Admin Page" : "everQuestionnaires"}</h1>
          <p className="lead text-muted">
            Here for your questionnaire needs anytime anywhere
          </p>
          <div className="top-buttons">
            <Link to='/questionnaires/new' className="btn btn-outline-info">
              <h4>Create A Questionnaire</h4>
            </Link>
          </div>
      { !isAdminPage &&
          <div className="top-buttons">
            <Link to='/admin/questionnaires' className="btn btn-outline-info">
              <h4>View Answers To Your Questionnaires</h4>
            </Link>
          </div> }
        </div>
      </section>
        {questionnaires && <QuestionnairesList questionnaires={questionnaires} url={url} isAdminPage={isAdminPage} />}
    </div>
  );
}

Questionnaires.propTypes = {
  questionnaires: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
  }).isRequired),
  url: PropTypes.string
};

export default Questionnaires;