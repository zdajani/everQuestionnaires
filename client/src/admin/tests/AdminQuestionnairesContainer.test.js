import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import ConnectedAdminQuestionnairesContainer, { AdminQuestionnairesContainer } from '../containers/AdminQuestionnairesContainer';
import Questionnaires from '../../questionnaires/components/Questionnaires'
import Loading from '../../commonComponents/Loading';
import { formattedData } from './testData';

const middlewares = []
const mockStore = configureMockStore(middlewares);

const setup = (formattedData, isLoading) => {
  const match = {url: '/admin/questionnaires'}
  const container = shallow(
    <AdminQuestionnairesContainer 
      questionnaires={formattedData} 
      isLoading={isLoading}
      match={match}/>
  );
    
  return {
    container,
    match
  };
}

const connectedSetup = (store) => {
  const connectedContainer = shallow(
      <ConnectedAdminQuestionnairesContainer store={store}/>
  );
  return {
    connectedContainer
  };
}

describe('AdminQuestionnaire container', () => {
  it('renders without crashing', () => {  
    const store = mockStore({
      adminQuestionnaires: {
        isLoading: true, 
        data: null,
        errorMessage: null
      },
    });
    
    const fetchAdminQuestionnairesSpy = jest.fn();
    const match = {url: '/admin/questionnaires'}
    
    const mountedComponent = mount(<AdminQuestionnairesContainer 
      store={store} 
      isLoading={true} 
      fetchAdminQuestionnaires={fetchAdminQuestionnairesSpy} 
      match={match} />);
    
    expect(toJson(mountedComponent)).toMatchSnapshot();
  });
  
  describe('when questionnaires data is available', () => {  
    const store = mockStore({
      adminQuestionnaires: {
        isLoading: false, 
        data: formattedData,
        errorMessage: null
      },
    });
    
    it('renders self with questionnaires props', () => {   
      const { connectedContainer } = connectedSetup(store); 
      expect(connectedContainer).toHaveProp('questionnaires', formattedData);
      expect(toJson(connectedContainer)).toMatchSnapshot();
    });
    
    it('renders Questionnaires component', () => {   
      const isloading = false; 
      const { container } = setup(formattedData, isloading);    
      const questionnaires = container.find(Questionnaires)
      expect(toJson(questionnaires)).toMatchSnapshot();
    });
    
    it('does not render Loading component', () => {    
      const { container } = setup(formattedData);       
  
      expect(container).not.toContainReact(<Loading />);
    });
  });
  
  describe('when loading', () => {
    const data = null;
    const isloading = true;     
    const { container } = setup(data, isloading); 
    
    it('renders loading component', () => {  
      expect(container).toContainReact(<Loading />);
    });
    
    it('does not render QuestionnairesList component', () => {  
      expect(container).not.toContainReact(
        <Questionnaires questionnaires={formattedData} url='/admin/questionnaires' />
      );
    });
  });
});