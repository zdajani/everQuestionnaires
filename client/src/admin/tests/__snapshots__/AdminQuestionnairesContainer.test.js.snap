// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminQuestionnaire container renders without crashing 1`] = `
<AdminQuestionnairesContainer
  fetchAdminQuestionnaires={[Function]}
  isLoading={true}
  match={
    Object {
      "url": "/admin/questionnaires",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <Loading>
      <div
        className="loading-screen"
      >
        <img
          alt="loadig"
          id="loading"
          src="https://tools.cisco.com/security/center/Resources/images/loader.gif"
        />
      </div>
    </Loading>
  </div>
</AdminQuestionnairesContainer>
`;

exports[`AdminQuestionnaire container when questionnaires data is available renders Questionnaires component 1`] = `
<Questionnaires
  questionnaires={
    Object {
      "1": Object {
        "id": 1,
        "title": "Cat breeds",
      },
      "2": Object {
        "id": 2,
        "title": "Cat breeds Questionnaire Part 2",
      },
    }
  }
  url="/admin/questionnaires"
/>
`;

exports[`AdminQuestionnaire container when questionnaires data is available renders self with questionnaires props 1`] = `
<AdminQuestionnairesContainer
  fetchAdminQuestionnaires={[Function]}
  isLoading={false}
  questionnaires={
    Object {
      "1": Object {
        "id": 1,
        "title": "Cat breeds",
      },
      "2": Object {
        "id": 2,
        "title": "Cat breeds Questionnaire Part 2",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
